six/modules/c++/six/source/NITFReadControl.cpp

```diff
@@ -81,21 +81,112 @@ six::PixelType getPixelType(nitf::ImageSubheader& subheader)
 
 namespace six
 {
+DataType NITFReadControl::getDataType(nitf::Record& record)
+{
+    nitf::List des = record.getDataExtensions();
+    if (des.isEmpty())
+    {
+        return DataType::NOT_SET;
+    }
+    nitf::DESegment segment = (nitf::DESegment) des.getFirst().getData();
```

Comment:
Add a comment that the SICD spec guarantees that the first DES will be the SICD/SIDD DES